src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/_assets.py

```diff
@@ -22,16 +24,54 @@ def _get_hip_file() -> str:
     return hou.hipFile.path()
 
 
+def _houdini_time_vars_to_glob(str_: str) -> str:
+    """
+    Given a filename, replace any houdini time-based variables or expressions with globs
+    to capture complete file sequences for job attachments. If there is no time-based
+    parameterization, then the original string is returned.
+
+    These are the useful timebased variables that can be included in files.
+    ref: https://www.sidefx.com/docs/houdini/render/expressions.html
+    """
+    time_based_variable_patterns: tuple = (
+        "(`.*`)",  # an expression, like python!
+        "(\$FF)",  # current fractional frame number
+        "(\${FF})",
+        "(\$F\d*)",  # current frame number, only variable with digits that follow
+        "(\${F\d*})",
+        "(\$T)",  # current time
+        "(\${T})",
+        "(\$SF)",  # current simulation timestep number
+        "(\${SF})",
+        "(\$ST)",  # current simulation time
+        "(\${ST})",
+    )
+    time_var_pattern = "|".join(time_based_variable_patterns)
+
+    return re.sub(time_var_pattern, "*", str_)
+
+
 def _get_asset_references(rop_node: hou.Node) -> AssetReferences:
     """
     Get the current paths stored in the parms backing the UI and return them as
     an AssetReferences object
     """
     asset_references = AssetReferences()
 
-    asset_references.input_filenames.update(
-        [n.eval() for n in rop_node.parm("input_filenames").multiParmInstances()]
-    )
+    for n in rop_node.parm("input_filenames").multiParmInstances():
+        filepath = n.unexpandedString()
+        filepath_glob = _houdini_time_vars_to_glob(filepath)
+        if filepath == filepath_glob:
+            asset_references.input_filenames.add(n.eval())
+            continue
+
+        # This is a work-around to avoid evaluating the offsets for
+        # scene introspection. It's possible that this will capture
+        # extra files since a glob will not enforce numbers, but
+        # handles well-defined file path patterns
+        for file in glob.iglob(filepath_glob):
+            asset_references.input_filenames.add(file)
+
     asset_references.input_directories.update(
```

Comment:
We don't expand tokens in directory names?

Reply:
This is addressed by a comment in the PR description

Reply:
Ah missed the comment. So there is concern about how to handle directory traversal that may happen as part of the expansion. What is the expectation then for users in terms of tokens in directories then, is that a complete non-starter and it shouldn't be done at all?

Reply:
My current take on it, is that it's more rare than being in filenames - though I'm sure it'll come up eventually. To avoid an absolute explosion of files being added (ie. if the last folder is just the token) we actually have to pattern match the results to the token after globbing.

I'll put a thing in the FAQ about this

Reply:
That sounds good to me. If it becomes an issue for people and is reported we can revisit.